### PR TITLE
added search customer by email api

### PR DESCRIPTION
**Pull request by PFadel**
See the example:
getCustomersByEmail.js

Features:

Added get method
Url encode emails

updates from: https://github.com/customerio/customerio-node/pull/89

### CODE REVIEW

File: lib/api.ts [hunk 1]
```diff
@@ -51,6 +52,14 @@ export class APIClient {
     return this.request.post(`${this.apiRoot}/send/email`, req.message);
   }
 
+  getCustomersByEmail(email: string) {
+    if (typeof email !== 'string' || !email) {
```
Comment (mike-engel) [inline lib/api.ts:56]: We prefer the [`isEmpty`](https://github.com/customerio/customerio-node/blob/c064c61dc2131f37a7d08c33ac0588ac9884295e/lib/utils.ts#L3) util to the `!` operator to catch cases where the value is an empty string

File: test/api.ts [hunk 2]
```diff
@@ -133,6 +133,19 @@ test('#sendEmail: override body: success', (t) => {
...
 
+test('#getCustomersByEmail: searching for a customer email (default)', (t) => {
+  sinon.stub(t.context.client.request, 'get');
+
+  const email = 'hello@world.com';
+  t.context.client.getCustomersByEmail(email);
+  t.truthy((t.context.client.request.get as SinonStub).calledWith(`${RegionUS.apiUrl}/customers?email=${email}`));
+})
+
+test('#getCustomersByEmail: should throw error when email is empty', (t) => {
```
Comment (mike-engel) [inline test/api.ts:144]: Another test should be added for cases where `null`/`undefined` or a non-string object are used for non-typescript clients

**Review by mike-engel — CHANGES_REQUESTED** [review]

File: lib/api.ts (hunk 1 above)
Comment (PFadel) [reply lib/api.ts]: done!

File: test/api.ts (hunk 2 above)
Comment (PFadel) [reply test/api.ts:144]: done!

**Comment by PFadel** [comment]
@mike-engel could you take another look?

**Review by mike-engel — COMMENTED** [review]
Thanks @PFadel! One last thing I forgot—can you add documentation to the readme?

**Comment by PFadel** [comment]
> Thanks @PFadel! One last thing I forgot—can you add documentation to the readme?

@mike-engel done!

**Comment by PFadel** [comment]
@mike-engel could you take another look? <3

**Comment by mike-engel** [comment]
Thanks for your patience @PFadel. This was included with 3.4.0
